compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -8619,6 +8619,9 @@ private BType getLaxFieldAccessType(BType exprType) {
                 return symTable.jsonType;
             case TypeTags.XML:
             case TypeTags.XML_ELEMENT:
+            case TypeTags.XML_COMMENT:
```

Comment:
My concern is why we include XML within the `getLaxFieldAccessType` function. If the type is lax, then the lax field access rules are included. XML is not a lax type, correct?

Reply:
As per the offline discussion, even though xml is not considered as lax type, lax typing rules apply for xml attribute access. I guess that is the reason for having xml type here.